File: app/src/test/java/com/google/android/stardroid/units/LatLong2Test.java
// Copyright 2008 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.android.stardroid.units;

import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;

public class LatLong2Test {
  private static final float TOL = 1e-4f;

  @Test
  public void testDistanceFrom90Degrees() {
    LatLong l1 = new LatLong(0, 0);
    LatLong l2 = new LatLong(0, 90);
    assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(90f);
  }

  @Test
  public void testDistanceFromSame() {
    LatLong l1 = new LatLong(30, 9);
    LatLong l2 = new LatLong(30, 9);
    assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(0f);
  }

  @Test
  public void testDistanceFromOppositePoles() {
    LatLong l1 = new LatLong(-90, 45);
    LatLong l2 = new LatLong(90, 45);
    assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(180f);
  }

  @Test
  public void testDistanceFromOnEquator() {
    LatLong l1 = new LatLong(0, -20);
    LatLong l2 = new LatLong(0, 30);
    assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(50f);
  }

  @Test
  public void testDistanceFromOnMeridian() {
    LatLong l1 = new LatLong(-10, 0);
    LatLong l2 = new LatLong(40, 0);
    assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(50f);
  }
}

File: app/src/test/java/com/google/android/stardroid/units/LatLong2Test.kt
// Copyright 2008 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
package com.google.android.stardroid.units

import com.google.common.truth.Truth.assertThat
import org.junit.Test

class LatLong2Test {
    @Test
    fun testDistanceFrom90Degrees() {
        val l1 = LatLong(0f, 0f)
        val l2 = LatLong(0f, 90f)
        assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(90f)
    }

    @Test
    fun testDistanceFromSame() {
        val l1 = LatLong(30f, 9f)
        val l2 = LatLong(30f, 9f)
        assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(0f)
    }

    @Test
    fun testDistanceFromOppositePoles() {
        val l1 = LatLong(-90f, 45f)
        val l2 = LatLong(90f, 45f)
        assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(180f)
    }

    @Test
    fun testDistanceFromOnEquator() {
        val l1 = LatLong(0f, -20f)
        val l2 = LatLong(0f, 30f)
        assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(50f)
    }

    @Test
    fun testDistanceFromOnMeridian() {
        val l1 = LatLong(-10f, 0f)
        val l2 = LatLong(40f, 0f)
        assertThat(l1.distanceFrom(l2)).isWithin(TOL).of(50f)
    }

    companion object {
        private const val TOL = 1e-4f
    }
}